### ISSUE TITLE
Vehicle fails to spawn on some newly-settled maps

### ISSUE THREAD

**Issue by Bqr1s** [SmashPhil/Vehicle-Framework#186]
**Describe the bug**
Vehicle fails to spawn on newly-settled map decently frequently. 2 out of 6 times (different results due to map randomness) failed with this error:
https://github.com/SmashPhil/Vehicle-Framework/blob/91c32ecb005fec0b831e02745828f24277d7108e/Source/Vehicles/Harmony/PatchCategories/Construction.cs#L20

**To Reproduce**
1. Have a 7x7 tiles vehicle (boarded with pawns) on the world map in free world tile. Superheavy shuttle from SOS 2 for example.
2. Have 2-sides coast Geological landform form respective mod in that tile. Such as southeastern, but not just southern. The issue reproduces with terrain flat, biome: arid shrubland.
4. Click "Settle". It might require several attempts to reproduce.
5. Vehicle fails to spawn with this message:
https://github.com/SmashPhil/Vehicle-Framework/blob/91c32ecb005fec0b831e02745828f24277d7108e/Source/Vehicles/Harmony/PatchCategories/Construction.cs#L202
Because that failure happens during map generation, the result of this issue is return to main menu with error message.
**Screenshots**
![screenshot129](https://github.com/user-attachments/assets/795152f9-9495-4281-b17c-6c578c6ed67a)
This screenshot is an example of successful map creation, as have nothing to show on fail

**Additional context**
For what I understand, this error should happen either super rarely or on "broken" maps like fully underground/completely covered by mountain or buildings or maybe for really huge vehicles, but not on a normal map with reasonable sized vehicle.

With 469 000+ Steam subscription, Geological landforms mod looks super common, not like some exotic mod that causes conflicts.

**Comment by SmashPhil** [comment]
This wouldn't be due to any mod conflicts.  Larger vehicles have issues with finding valid spots in general just because they take up much more room, but in this particular case it's probably just picking a bad spot and not finding any alternative in the immediate area.  Shouldn't be too difficult to fix

**Comment by Bqr1s** [comment]
Yes. I think I've forgot to attach full log, can look for that needed.
